libp2p/peer.ss

```diff
@@ -39,11 +39,22 @@
   (pregexp "/ipfs/"))
 
 (def (string->peer-info str)
-  (match (pregexp-split ipfs-part-rx str)
-    ([maddr id]
-     (peer-info (string->ID id)
-                (if (string-empty? maddr)
-                  []
-                  [(string->multiaddr maddr)])))
+  (match (reverse (pregexp-split ipfs-part-rx str))
+    ([id . rest]
+     (peer-info (if (equal? rest '())
+                  (error "Malformed peer info" str)
+                  (string->ID id))
+                (let ((maddr (string-append-with-ipfs (reverse rest))))
+                  (if (string-empty? maddr)
+                    []
+                    [(string->multiaddr maddr)]))))
     (else
      (error "Malformed peer info" str))))
+
+(def (string-append-with-ipfs lst)
```

Comment:
Can this whole routine be replaced with a call to `string-join`?

Reply:
yeah, lets string-join